src/plugins/twitter/twitter.py

```diff
@@ -1,10 +1,135 @@
+from collections.abc import Iterable
+from datetime import datetime
+from logging import getLogger
+
+from TwitterSearch import TwitterSearchOrder
+
+from flask import current_app
+
 from generic.mount_point import GenericDataProvider
+from processor.logic import set_generic_twitter_link
+from processor.models import Event, RawEvent
+
+
+logger = getLogger(__name__)
 
 
 class TwitterProvider(GenericDataProvider):
 
-    def load(self, string):
-        pass
+    def instantiate_client(self):
+        try:
+            return self.client_class(
+                consumer_key=current_app.config.get('TWITTER_APP_KEY'),
+                consumer_secret=current_app.config.get(
+                    'TWITTER_APP_KEY_SECRET'
+                ),
+                access_token=current_app.config.get('TWITTER_ACCESS_TOKEN'),
+                access_token_secret=current_app.config.get(
+                    'TWITTER_ACCESS_TOKEN_SECRET'
+                ),
+            )
+        except RuntimeError:
+            logger.error('App not found - skipping client instantiation')
+            return None
+
+    def _validate(self, results):
+        return self.validator.dump(results)
+
+    @staticmethod
+    def _convert_to_python(event_entry):
+        """ Parse data from twitter API and convert them to Python types."""
+        date_string = event_entry.pop('created_at_str')
+        created_at = datetime.strptime(date_string, '%a %b %d %X %z %Y')
+
+        twitter_id = event_entry.pop('twitter_id')
+        subject_id = set_generic_twitter_link(twitter_id)
+
+        return subject_id, created_at
+
+    def _build(self, event_data, uri_id, origin, event_dict):
+        """ Build a Event objects using the defined schema.
```

Comment:
Minor: Build _an_ event object?

Reply:
I knew one day the English language would turn on me. 
But, yeah, I think it returns a number of objects so we can drop the `'a'` altogether :thinking: